### PR TITLE
[Chore](execution) change PipelineTaskState to enum class && remove some row-based code

### PR DESCRIPTION
**Pull request by BiteTheDDDDt**
# Proposed changes

1. change PipelineTaskState to enum class
2. remove some row-based code on FoldConstantExecutor::_get_result
3. reduce memcpy on minmax runtime filter function(Now we can guarantee that the input data is aligned)
4. add Wunused-template check, and remove some unused function, change some static function to inline function.

## Problem summary

Describe your changes.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/vec/functions/cast_type_to_either.h [hunk 1]
```diff
@@ -27,7 +27,7 @@ namespace doris::vectorized {
 class IDataType;
 
 template <typename... Ts, typename F>
-static bool cast_type_to_either(const IDataType* type, F&& f) {
+inline bool cast_type_to_either(const IDataType* type, F&& f) {
```
Comment (carlvinhust2012) [inline be/src/vec/functions/cast_type_to_either.h:30]: here change from 'static' to 'inline' , does not affect the visible range of the function?

Comment (BiteTheDDDDt) [reply be/src/vec/functions/cast_type_to_either.h:30]:
```
inline marks the symbol WEAK, which hints linker to choose arbitary one of definition in object files
static marks the symbol LOCAL, which restricts the symbol in current translation unit, and the linker may keep multiple instances of (possibly different) definition.
```
https://gist.github.com/htfy96/50308afc11678d2e3766a36aa60d5f75

A header containing a static function is included by other files, which may generate an unused-template

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

File: be/src/vec/functions/cast_type_to_either.h (hunk 1 above)
Comment (HappenLee) [reply be/src/vec/functions/cast_type_to_either.h:30]:
```For template functions, I'd say that using static, inline or no keyword produce the same result; in fact, in all cases the function will be inlined and no multiple definition error will be raised.
```
https://stackoverflow.com/questions/29495678/why-use-static-function-template

File: be/src/runtime/fold_constant_executor.cpp [hunk 2]
```diff
@@ -180,24 +179,14 @@ string FoldConstantExecutor::_get_result(void* src, size_t size, const TypeDescr
...
     case TYPE_OBJECT: {
-        if constexpr (is_vec) {
-            return std::string((char*)src, size);
-        }
-        return (reinterpret_cast<StringRef*>(src))->to_string();
+        return std::string((char*)src, size);
     }
     case TYPE_DATE:
     case TYPE_DATETIME: {
-        if constexpr (is_vec) {
```
Comment (HappenLee) [inline be/src/runtime/fold_constant_executor.cpp:190]: better remove `is_vec` ？

Comment (BiteTheDDDDt) [reply be/src/runtime/fold_constant_executor.cpp:190]: has been removed

**Comment by BiteTheDDDDt** [comment]
run buildall

**Review by HappenLee — DISMISSED** [review]
LGTM

**Comment by BiteTheDDDDt** [comment]
run arm

**Comment by BiteTheDDDDt** [comment]
run p0

**Comment by BiteTheDDDDt** [comment]
run p0

**Comment by BiteTheDDDDt** [comment]
run arm

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run COMPILE

**Comment by BiteTheDDDDt** [comment]
run compile

**Review by HappenLee — APPROVED** [review]
LGTM

**Comment by BiteTheDDDDt** [comment]
run p0

**Comment by BiteTheDDDDt** [comment]
run p0

**Comment by BiteTheDDDDt** [comment]
run p0

**Pull request merged**